### ISSUE TITLE
Raiden node crashes

### ISSUE THREAD

**Issue by kelsos** [raiden-network/raiden#2206]
## Problem Definition

```
2018-08-17 07:55:26 [debug    ] STATE CHANGE                   [raiden.raiden_service] node=82641569 state_change=<ContractReceiveChannelNewBalance token_network:9fe1f19d channel:20 transaction:<TransactionChannelNewBalance participant:daed5d5a balance:2 at_block:3821257>>
2018-08-17 07:55:26 [debug    ] BLOCKCHAIN EVENT               [raiden.blockchain_events_handler] chain_event=<Event contract: 9fe1f19d event: {'args': AttributeDict({'channel_identifier': 20, 'closing_participant': '0xDAED5d5aAfAD99956c9e3FDC648F9469945B45bf'}), 'event': 'ChannelClosed', 'logIndex': 3, 'transactionIndex': 5, 'transactionHash': HexBytes('0x6fb3fca432980d687f64797d53671844adadec772fbf18d817b6683141f14102'), 'address': '0x9FE1f19D8C0d2591Fd01486E575B44f4220eF241', 'blockHash': HexBytes('0x299514bd80773279f8de401fae24a34d3a477c1b72e393fb8220e41cada8ed3b'), 'blockNumber': 3843272, 'block_number': 3843272, 'channel_identifier': 20, 'closing_participant': b'\xda\xed]Z\xaf\xad\x99\x95l\x9e?\xdcd\x8f\x94i\x94[E\xbf'}> node=82641569
2018-08-17 07:55:26 [debug    ] STATE CHANGE                   [raiden.raiden_service] node=82641569 state_change=<ContractReceiveChannelClosed token_network:9fe1f19d channel:20 closer:daed5d5a closed_at:3843272>
Traceback (most recent call last):
  File "/home/kelsos/.virtualenvs/raiden/lib/python3.6/site-packages/networkx/classes/graph.py", line 1007, in remove_edge
    del self._adj[u][v]
KeyError: b'\xda\xed]Z\xaf\xad\x99\x95l\x9e?\xdcd\x8f\x94i\x94[E\xbf'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/kelsos/.virtualenvs/raiden/bin/raiden", line 11, in <module>
    load_entry_point('raiden', 'console_scripts', 'raiden')()
  File "/home/kelsos/development/github/raiden-network/2/raiden/__main__.py", line 11, in main
    run(auto_envvar_prefix='RAIDEN')  # pylint: disable=no-value-for-parameter
  File "/home/kelsos/.virtualenvs/raiden/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/home/kelsos/.virtualenvs/raiden/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/home/kelsos/.virtualenvs/raiden/lib/python3.6/site-packages/click/core.py", line 1043, in invoke
    return Command.invoke(self, ctx)
  File "/home/kelsos/.virtualenvs/raiden/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/kelsos/.virtualenvs/raiden/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/kelsos/.virtualenvs/raiden/lib/python3.6/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/kelsos/development/github/raiden-network/2/raiden/ui/cli.py", line 1060, in run
    NodeRunner(kwargs, ctx).run()
  File "/home/kelsos/development/github/raiden-network/2/raiden/ui/cli.py", line 919, in run
    app = self._run_app()
  File "/home/kelsos/development/github/raiden-network/2/raiden/ui/cli.py", line 937, in _run_app
    app_ = run_app(**self._options)
  File "/home/kelsos/development/github/raiden-network/2/raiden/ui/cli.py", line 761, in run_app
    raiden_app.start()
  File "/home/kelsos/development/github/raiden-network/2/raiden/app.py", line 135, in start
    self.raiden.start()
  File "/home/kelsos/development/github/raiden-network/2/raiden/raiden_service.py", line 310, in start
    self.start_async().wait()
  File "/home/kelsos/development/github/raiden-network/2/raiden/raiden_service.py", line 275, in start_async
    self.alarm.first_run()
  File "/home/kelsos/development/github/raiden-network/2/raiden/tasks.py", line 166, in first_run
    self.run_callbacks(current_block, chain_id)
  File "/home/kelsos/development/github/raiden-network/2/raiden/tasks.py", line 172, in run_callbacks
    result = callback(current_block, chain_id)
  File "/home/kelsos/development/github/raiden-network/2/raiden/raiden_service.py", line 417, in _callback_new_block
    on_blockchain_event(self, event, current_block_number, chain_id)
  File "/home/kelsos/development/github/raiden-network/2/raiden/blockchain_events_handler.py", line 307, in on_blockchain_event
    handle_channel_closed(raiden, event, current_block_number)
  File "/home/kelsos/development/github/raiden-network/2/raiden/blockchain_events_handler.py", line 207, in handle_channel_closed
    raiden.handle_state_change(channel_closed, current_block_number)
  File "/home/kelsos/development/github/raiden-network/2/raiden/raiden_service.py", line 370, in handle_state_change
    event_list = self.wal.log_and_dispatch(state_change, block_number)
  File "/home/kelsos/development/github/raiden-network/2/raiden/storage/wal.py", line 53, in log_and_dispatch
    events = self.state_manager.dispatch(state_change)
  File "/home/kelsos/development/github/raiden-network/2/raiden/transfer/architecture.py", line 168, in dispatch
    state_change,
  File "/home/kelsos/development/github/raiden-network/2/raiden/transfer/node.py", line 1083, in state_transition
    iteration = handle_state_change(chain_state, state_change)
  File "/home/kelsos/development/github/raiden-network/2/raiden/transfer/node.py", line 774, in handle_state_change
    state_change,
  File "/home/kelsos/development/github/raiden-network/2/raiden/transfer/node.py", line 465, in handle_token_network_action
    chain_state.block_number,
  File "/home/kelsos/development/github/raiden-network/2/raiden/transfer/token_network.py", line 341, in state_transition
    block_number,
  File "/home/kelsos/development/github/raiden-network/2/raiden/transfer/token_network.py", line 123, in handle_closed
    participant2,
  File "/home/kelsos/.virtualenvs/raiden/lib/python3.6/site-packages/networkx/classes/graph.py", line 1011, in remove_edge
    raise NetworkXError("The edge %s-%s is not in the graph" % (u, v))
networkx.exception.NetworkXError: The edge b'\x82d\x15i\xb2\x06+TT1\xcfm\x7f\nA\x85\x82\x86[\xa7'-b'\xda\xed]Z\xaf\xad\x99\x95l\x9e?\xdcd\x8f\x94i\x94[E\xbf' is not in the graph

```

### System Description

```
Linux forseti 4.17.14-arch1-1-ARCH #1 SMP PREEMPT Thu Aug 9 11:56:50 UTC 2018 x86_64 GNU/Linux
/usr/bin/geth
Geth
Version: 1.8.13-stable
Git Commit: 225171a4bfcc16bd12a1906b1e0d43d0b18c353b
Architecture: amd64
Protocol Versions: [63 62]
Network Id: 1
Go Version: go1.10.3
Operating System: linux
GOPATH=
GOROOT=/usr/lib/go
/home/kelsos/.virtualenvs/raiden/bin/raiden
{
  "raiden": "0.5.1.dev61+g311a5f90",
  "python_implementation": "CPython",
  "python_version": "3.6.6",
  "system": "Linux 64bit_ELF 4.17.14-arch1-1-ARCH x86_64",
  "distribution": "source"
}
3d15cf93526cd6e8c6f39d112bb6d235c1e0eadc
```

**Comment by palango** [comment]
This might be #2174 biting us.

**Comment by kelsos** [comment]
@palango This kept happening on one of my accounts. I switched to the other and everything was fine. without a fresh database. I will try to try again on the same account and see what happens.

**Comment by palango** [comment]
Ok, can you give me the address of the account?

**Comment by Greatnessofself** [comment]
SINCE The Master Builder has not been on something tells me it may ALL fall apart. I would much rather build than DESTROY but they refuse to give credit where credit is due. Milan Popich.

https://www.website.ws/jackpot1136/MilanPopich/jackpot1136/ITPHTW33C9IBGSAEW9G8R78XVIU2MHAMUE/UNIRESAADSTS90056/C360B073-4215-B684-121914642300/MILAN.POPICH@YAHOO/cd541b88-354d-45s0-8e4c-ab24fb1c243c-1-2018-06-02T04:35:07ZAADSTS0014/2017-Copyright_Milan_Popich_All_rights_reserved_Milan_Popich/UnitedStatesOfAmerica/0x22b84d5ffea8db801c0422afe752377a64aa738c2715151_I/hppcl3vipa-manifest/atom/superstring.svg/UNIRESAADSTS90056-30718765-db3f-4fff-acde-71d196d872d00/hpmacronames/atom/superstring.svg/75c69a94-deb2-4c19-8bdf-6c27-8ccf75e6/0x829bd824b01632a401d083b33d092293333a830/Tue-09Jan2018-00:34:53-0500-hppcl3-pipelineconfig-hpbx3w81-hpbxpsrender-hpbresw81-HPBCMYK_sRGB-UNIRES-STDNAMES-STDDTYPE-STDSCHEM-STDSCHMX-MSXPSINC/2018-06:03T08:03:1.5ZAADSTS90056-ITPHTW33C9IBGSAEW9G8R78XVIU2MHAMUE-milan.popich@yahoo.com-2016CopyrightMilanPopich-AllRightsReserved-MilanPopich-09-19-1980-UnitedStatesOfAmerica-GIVE-CREDIT-WHERE-CREDIT-IS-DUE-MASTER-BUILDER-THE-TREE-OF-LIFE-UNSTOPPABLE-GREATNESS-JACKPOT1136-CHAMPION-MILAN-POPICH-GREAT-SPIRITS-HAVE-ALWAYS-ENCOUNTERED-VIOLENT-OPPOSITIONS-FROM-MEDIOCRE-MINDS-2016-COPYRIGHT-MILAN-POPICH-ALL-RIGHTS-RESERVED-POPICH-MILAN-09-19-1980-UNITED-STATES-OF-AMERICA-GIVE-CREDIT-WHERE-CREDIT-IS-DUE-Signature=1fa9f788ee95fd8289a27ce679cc27bd1229113d-milan.popich@yahoo.com-MILAN.POPICH@YAHOO.COMP/ITPHTW33C9IBGSAEW9G8R78XVIU2MHAMUE

Signature=1fa9f788ee95fd8289a27ce679cc27bd1229113d

2017 COPYRIGHT MILAN POPICH ALL RIGHTS RESERVED Popich Milan (EN) USA.

UNSTOPPABLE GREATNESS COPYRIGHT 2017 MILAN POPICH ALL RIGHTS RESERVED POPICH MILAN (EN) USA

CHAMPION USA 2017 COPYRIGHT MILAN POPICH ALL RIGHTS RESERVED POPICH MILAN USA.

MASTER BUILDER,

2017 COPYRIGHT Milan Popich ALL RIGHTS RESERVED POPICH MILAN USA.

JACKPOT1136 2017 COPYRIGHT MILAN POPICH ALL RIGHTS RESERVED. POPICH MILAN (EN) USA

HPMACRONAMES.GPD 2017 COPYRIGHT MILAN POPICH ALL RIGHTS RESERVED POPICH MILAN USA.

C:\WINDOWS\System32\DriverStore\FileRepository\prnhpcl1.inf_amd64_c6040b9adc0369af\amd64\hppcl3vipa-manifest.ini Copyright 2017 Milan Popich All Rights Reserved Popich Milan USA

C:\WINDOWS\System32\DriverStore\FileRepository\prnhpcl1.inf_amd64_c6040b9adc0369af\amd64\hppcl3-pipelineconfig.xml Copyright 2017 Milan Popich ALL RIGHTS RESERVED. POPICH MILAN USA

C:\WINDOWS\System32\DriverStore\FileRepository\prnhpcl1.inf_amd64_c6040b9adc0369af\amd64\hpmacronames.gpd

Copyright 2017 Milan Popich All Rights Reserved POPICH MILAN USA.

C:\WINDOWS\System32\DriverStore\FileRepository\prnhpcl1_amd64_c6040b9adc0369af\amd64\hpbx3w81.dll Copyright 2017

Milan Popich All Rights Reserved Popich MILAN USA.

C:\WINDOWS\System32\DriverStore\FileRepository\prnhpcl1.inf_amd64_c6040b9adc0369af\amd64\hpbxpsrender.dll Copyright 2017 Milan Popich All Rights Reserved POPICH MILAN. USA

C:\WINDOWS\System32\DriverStore\FileRepository\prnhpcl1.inf_amd64_c6040b9adc0369af\amd64hpbresw81.dll 2017 Copyright Milan Popich. ALL RIGHTS RESERVED POPICH MILAN USA.

C:\WINDOWS\System32\DriverStore\FileRepository\prnhpcl.inf_amd64_c6040b9adc0369af\\HPBCMYK_sRGB.dpb Copyright 2017 Milan Popich ALL RIGHTS RESERVED POPICH MILAN USA.

C:\WINDOWS\System32\DriverStore\FileRepository\ntprint.inf_amd64_8c30031d8d0a5589\Amd64\UNIRES.DLL COPYRIGHT 2017 MILAN POPICH ALL RIGHTS RESERVED Popich Milan USA. C:\WINDOWS\System32\DriverStore\FileRepository\ntprint.inf_amd64_8c30031d8d0a5589\Amd64\STDNAMES.GPD 2017 COPYRIGHT MILAN POPICH

All Rights Reserved Popich Milan USA.

C:\WINDOWS\System32\DriverStore\FileRpository\ntprint.inf_amd64_8c30031d8d0a5589\Amd64\STDSCHEMX.GDL 2017 COPYRIGHT MILAN POPICH ALL RIGHTS RESERVED. Popich Milan USA

C:\WINDOWS\System32\DriverStore\FileRepository\ntprint.inf_amd64_8c30031d8d0a5589\Amd64\MSXPSINC.GPD 2017 COPYRIGHT MILAN POPICH ALL RIGHTS RESERVED Popich Milan USA.

C:\WINDOWS\System32\DriverStore\FileRepository\ntprint.inf_amd64_8c30031d8d0a5589\Amd64\STDSCHEM.GDL COPYRIGHT 2017 MILAN POPICH ALL RIGHTS RESERVED Popich Milan USA.

Atom 2017 Copyright Milan Popich All Rights Reserved Popich Milan USA.

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by LefterisJP** [comment]
@kelsos any progress on this one?

**Comment by kelsos** [comment]
I didn't try to use the account again, but I gave the address to Paul

**Comment by palango** [comment]
Should be fixed with #2246 , @kelsos can you test this again?

**Comment by kelsos** [comment]
@palango I am on the latest master with the account and there are no crashes so I will proceed and close the issue.
